src/libs/tmdk/src/commands/deploy.ts

```
@@ -101,108 +100,102 @@ export const handler = async (argv: Arguments<Options>) => {
+        "propertyDefinitions"
+      ] as object;
+      if (propertyDefinitions != undefined) {
+        const filtered_property_definitions = Object.entries(
+          propertyDefinitions
+        ).reduce((acc, [key, value]) => {
+          if (!value["isInherited"]) {
+            acc[key] = value;
           }
-        }
+          return acc;
+        }, {} as { [key: string]: object });
+        componentTypeDefinition["propertyDefinitions"] =
+          filtered_property_definitions;
       }
-    }
-  }
-
-  // import scenes
-  if (tmdk_config.hasOwnProperty("scenes")) {
-    console.log("======== Scene Files ========");
-    for (const sceneFile of tmdk_config["scenes"]) {
-      console.log(`Importing scene: ${sceneFile} ...`);
+      // create component type if not exists
       try {
-        await importScene(
+        const alreadyExists: boolean = await createComponentTypeIfNotExists(
           workspaceIdStr,
-          `${dir}/${sceneFile}`,
-          workspaceContentBucket
+          componentTypeDefinition
         );
-      } catch (error) {
-        if (error instanceof ConflictException) {
+        await waitForComponentTypeActive(
+          workspaceIdStr,
+          componentTypeDefinition["componentTypeId"]
+        );
+        if (!alreadyExists) {
           console.log(
-            `  ...skipping scene creation for ${sceneFile} due to pre-existing scene with same id`
-          ); // TODO should scan and warn instead
+            `Created component-type: ${componentTypeDefinition["componentTypeId"]}`
+          );
+        }
+      } catch (e) {
+        if (
+          // components are not gaurenteed to be in the correct order; retry
+          e instanceof ValidationException &&
+          e.message.includes("do not exist in workspace")
+        ) {
+          stillComponentRemaining = true;
         } else {
-          throw error;
+          throw new Error(`Failed to create component type. ${e}`);
```

If there was an error we should throw the error for the full stack trace. Otherwise user will just see generic Error.
So, use form:
`console.error(Failed to create component type, ${e}");`
 `throw e;`

---

Interesting, good to know will update